Marksman/Toc.fs

```diff
@@ -0,0 +1,94 @@
+module Marksman.Toc
+
+open Marksman.Misc
+open Marksman.Index
+open Marksman.Cst
+open Marksman.Text
+open Ionide.LanguageServerProtocol.Types
+open Ionide.LanguageServerProtocol.Logging
+
+[<Literal>]
+let Marker = """<!--toc-->"""
```

Comment:
Can be configurable in `.marksman.toml` but I haven't looked into how to do that or whether it's useful